README.md

```diff
@@ -59,32 +63,139 @@ let deleteFile containerName blob  =
+[Manage blob properties and metadata with .NET](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-properties-metadata?tabs=dotnet)
+
+```fsharp
+let getProperties containerName blob =
+    let connString = "UseDevelopmentStorage=true"
+    let result = connString
+                    |> Blob.fromConnectionString
+                    |> Blob.container containerName
+                    |> Blob.getProperties blob
+                    |> Blob.execAsync<BlobProperties>
+    result
+	
+let getTestJsonPropertiesAndMetadata =
+	async {
+        let! blobProperites = getProperties "container" "test.json"  
+        let properties = blobProperites.Value
+        Console.WriteLine "Properties:"
+        Console.WriteLine ("  Content Type: " + properties.ContentType)
+        Console.WriteLine ("  Content Length: " + properties.ContentLength.ToString())
+        let metadata = blobProperites.Value.Metadata
+        Console.WriteLine "Metadata:"
+        metadata.Keys |> Seq.iter (fun k -> printfn "  \"%s\" : \"%s\"" k (metadata.Item k))            
+    } |> Async.RunSynchronously
+```
+
+### Set / Update Properties
+
+Note that when you wish to change a property you need to set all other values in
+the property object by first retrieving them by calling get properties.
+
+```fsharp
+let setProperties containerName blob properties =
+    let connString = "UseDevelopmentStorage=true"
+    let result = connString
+                    |> Blob.fromConnectionString
+                    |> Blob.container containerName
+                    |> Blob.setProperties blob properties
+                    |> Blob.execAsync<BlobInfo>
+    result
+	
+let setTestJsonProperties =
+    async {
+        let! blobProperites = getProperties "container" "test.json"  
+        let properties = new Azure.Storage.Blobs.Models.BlobHttpHeaders()
+        properties.CacheControl <- blobProperites.Value.CacheControl
+        properties.ContentDisposition <- blobProperites.Value.ContentDisposition
+        properties.ContentEncoding <- blobProperites.Value.ContentEncoding
+        properties.ContentHash <- blobProperites.Value.ContentHash
+        properties.ContentLanguage <- blobProperites.Value.ContentLanguage
+        properties.ContentType <- blobProperites.Value.ContentType
+        let! blobInfo = setProperties "container" "test.json" properties
+        Console.WriteLine "Properties set"
+    } |> Async.RunSynchronously
+```
+
+### Set / Update Metadata
+
+Note that when you wish to add or update metadata you need to first retrieve the
+existing metadata by calling get properties.
+
+```fsharp
+let setMetadata containerName blob metadata =
+    let connString = "UseDevelopmentStorage=true"
+    let result = connString
+                    |> Blob.fromConnectionString
+                    |> Blob.container containerName
+                    |> Blob.setMetadata blob metadata
+                    |> Blob.execAsync<BlobInfo>
+    result
+	
+let setTestJsonMetadata =
```

Comment:
In terms of setting metadata I think the closest to F# friendly API would be something like this:

```fsharp
 async {
        let newMetadata = 
            ["new-metadata-key", "new-metadata-value";
                "new-metadata-key2", "new-metadata-value2"]
            |> Map.ofList

        let! blobInfo = setMetadata "container" "test.json" newMetadata
        Console.WriteLine "Metadata set"
    } |> Async.RunSynchronously
```
reading and merging metadata with existing should be handled by the operation itself